lib/router.js

```diff
@@ -46,7 +46,7 @@ export default class Router {
 
   navigate(fragment, stateUpdate={}) {
     fragment = stripHash(fragment);
-    if (fragment === this.app.state.$fragment && !Object.keys(stateUpdate).length) {
+    if (!this.compiledRoutes.length || fragment === this.app.state.$fragment && !Object.keys(stateUpdate).length) {
```

Comment:
I would rather just not initialize/compile the routes/router in the first place rather than running a non-functional router if no routes are defined.